### PR TITLE
Feedback emails, ticket 6080, added Table 2: List of Tests, Certification Events, and Extension/Exemptions

### LINKED ISSUE #6080: Submission Confirmation Emails and Feedback Reports
During the 1/25/2024 weekly ECMPS Support meeting with EPA, there was a discussion on what to do with Feedback Report Maintenance in CAT.

The PDF feedback report that is included with 1.0 submission confirmation emails is not currently included in 2.0. This could be due to the new CDX process.

The only users that would have access to submission related information from CDX - Submission History would be the submitter themselves.

Need to determine if:

- It's even possible to include something similar to the PDF feedback report attachment in the submission confirmation emails.
- Or, just expand the data currently shown in the emailed Submission Confirmation Report.

### PR DESCRIPTION
**Pull request by yonatan-dp**
Feedback emails, ticket 6080, added Table 2: List of Tests, Certification Events, and Extension/Exemptions

### CODE REVIEW

File: deployments/ecmps/release-v2.0-fix/Sprint8/6080/camdecmps/get-qatest-feedback.sql [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    ts.test_type_cd,
+    ts.test_num,
+    COALESCE(c.component_identifier, ms.system_identifier, '') || '/' ||
+    COALESCE(c.component_type_cd || ' ' || COALESCE(ts.span_scale_cd, ''), ms.sys_type_cd, '') AS sys_comp,
+    CASE
+        WHEN rp.calendar_year IS NOT NULL THEN rp.period_abbreviation
+        ELSE camdecmps.format_date_hour(ts.end_date, ts.end_hour, ts.end_min)
+        END AS end_test
+FROM camdecmps.test_summary ts
+         INNER JOIN (
```
Comment (djw4erg) [inline deployments/ecmps/release-v2.0-fix/Sprint8/6080/camdecmps/get-qatest-feedback.sql:25]: The cl_filtered sub-query is not actually used in the query.  Is the purpose to make sure that a CHECK_LOG row exists for the associated Check Session?  If so, this method could use to duplicate rows for when more than one CHECK_LOG row exists.

File: deployments/ecmps/release-v2.0-fix/Sprint8/6080/camdecmps/get-qce-feedback.sql [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+BEGIN
+RETURN QUERY
+SELECT
+    COALESCE(ml.stack_name, ml.unitid) AS unit_stack_pipe,
+    qe.qa_cert_event_cd,
+    ms.system_identifier || '/' || ms.sys_type_cd AS sys_id_type,
+    c.component_identifier || '/' || c.component_type_cd AS component,
+    camdecmps.format_date_hour(qe.qa_cert_event_date, qe.qa_cert_event_hour, NULL) AS qa_cert_datehour
+FROM camdecmps.qa_cert_event qe
+         INNER JOIN (
```
Comment (djw4erg) [inline deployments/ecmps/release-v2.0-fix/Sprint8/6080/camdecmps/get-qce-feedback.sql:21]: See "deployments/ecmps/release-v2.0-fix/Sprint8/6080/camdecmps/get-qatest-feedback" comment on cl_filtered sub-query.

File: deployments/ecmps/release-v2.0-fix/Sprint8/6080/camdecmps/get-tee-feedback.sql [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+RETURN QUERY
+SELECT
+    COALESCE(ml.stack_name, ml.unitid) AS unit_stack_pipe,
+    tee.extens_exempt_cd,
+    rp.period_abbreviation AS tee_year_quarter,
+    ms.system_identifier || '/' || ms.sys_type_cd AS sys_id_type,
+    c.component_identifier || '/' || c.component_type_cd AS component,
+    COALESCE(tee.span_scale_cd, tee.fuel_cd) AS span_fuel
+FROM camdecmps.test_extension_exemption tee
+         INNER JOIN (
```
Comment (djw4erg) [inline deployments/ecmps/release-v2.0-fix/Sprint8/6080/camdecmps/get-tee-feedback.sql:23]: See "deployments/ecmps/release-v2.0-fix/Sprint8/6080/camdecmps/get-qatest-feedback" comment on cl_filtered sub-query.

File: deployments/ecmps/release-v2.0-fix/Sprint8/6080/camdecmps/vw_monitor_location.sql
Comment (djw4erg) [inline deployments/ecmps/release-v2.0-fix/Sprint8/6080/camdecmps/vw_monitor_location.sql:1]: I suggest the following would simplify the SQL for this view.  Also with the simplification, directly including the four tables in users of the view may be better/clearer than using the view.

Both are just suggestions.

```
SELECT  ml.mon_loc_id,
        p.oris_code,
        p.facility_name,
        COALESCE( sp.stack_name, u.unitid ) AS location_identifier,
        p.fac_id,
        p.state,
        p.county_cd,
        u.unit_id,
        sp.stack_pipe_id,
        u.non_load_based_ind,
        sp.active_date,
        sp.retire_date,
        u.unitid,
        sp.stack_name,
        u.comr_op_date,
        u.comm_op_date
 FROM   camdecmps.monitor_location ml
        LEFT JOIN camd.unit u 
          ON u.unit_id = ml.unit_id
        LEFT JOIN camdecmps.stack_pipe sp 
          ON sp.stack_pipe_id::text = ml.stack_pipe_id::text
        JOIN camd.plant p
          ON p.fac_id in ( u.fac_id, sp.fac_id );
```

File: deployments/ecmps/release-v2.0-fix/Sprint8/6080/camdecmps/get-qatest-feedback.sql (hunk 1 above)
Comment (yonatan-dp) [reply deployments/ecmps/release-v2.0-fix/Sprint8/6080/camdecmps/get-qatest-feedback.sql:25]: So, basically I replicated the existing query from SQL Server you provided to me. 

The relevant section of the original SQL Server  query is as follows (redacted)

FROM VW_RPT_FEEDBACK_CHECK_LOG cl
			INNER JOIN TEST_SUMMARY ts ON cl.CHK_SESSION_ID = ts.CHK_SESSION_ID 
			INNER JOIN QA_SUPP_DATA qa ON ts.TEST_SUM_ID = qa.TEST_SUM_ID 
			INNER JOIN VW_MONITOR_LOCATION ml ON ts.MON_LOC_ID = ml.MON_LOC_ID 
			LEFT OUTER JOIN COUNTY_CODE cc ON ml.COUNTY_CD = cc.COUNTY_CD 
			LEFT OUTER JOIN REPORTING_PERIOD rp ON ts.RPT_PERIOD_ID = rp.RPT_PERIOD_ID 
			LEFT OUTER JOIN COMPONENT c ON ts.COMPONENT_ID = c.COMPONENT_ID 
			LEFT OUTER JOIN MONITOR_SYSTEM ms ON ts.MON_SYS_ID = ms.MON_SYS_ID 
		WHERE (cl.SEVERITY_CD <> 'NONE' OR cl.SEVERITY_CD IS NULL)
		  AND ts.TEST_SUM_ID IN

I basically moved the VW_RPT_FEEDBACK_CHECK_LOG  FROM clause to an inner join. Note also that  this WHERE check that is in the inner join from the VW_RPT_FEEDBACK_CHECK_LOG  
   WHERE cl.severity_cd <> 'NONE' OR cl.severity_cd IS NULL
   
You also stated that... "If so, this method could use to duplicate rows for when more than one CHECK_LOG row exists."

I am not sure the implication of this, but my goal was to replicate the SQL Server logic as accurately as possible.

File: deployments/ecmps/release-v2.0-fix/Sprint8/6080/camdecmps/get-qce-feedback.sql (hunk 2 above)
Comment (yonatan-dp) [reply deployments/ecmps/release-v2.0-fix/Sprint8/6080/camdecmps/get-qce-feedback.sql:21]: See comment above...

File: deployments/ecmps/release-v2.0-fix/Sprint8/6080/camdecmps/get-tee-feedback.sql (hunk 3 above)
Comment (yonatan-dp) [reply deployments/ecmps/release-v2.0-fix/Sprint8/6080/camdecmps/get-tee-feedback.sql:23]: See comment above...

File: deployments/ecmps/release-v2.0-fix/Sprint8/6080/camdecmps/vw_monitor_location.sql
Comment (yonatan-dp) [reply deployments/ecmps/release-v2.0-fix/Sprint8/6080/camdecmps/vw_monitor_location.sql:1]: Have updated this. Have also added an update for camdecmpswks schema version of the view.

File: deployments/ecmps/release-v2.0-fix/Sprint8/6080/camdecmps/get-qatest-feedback.sql (hunk 1 above)
Comment (djw4erg) [reply deployments/ecmps/release-v2.0-fix/Sprint8/6080/camdecmps/get-qatest-feedback.sql:25]: Thanks for the clarification and I agree with your goal.  I am going to resolve this conversation.

File: deployments/ecmps/release-v2.0-fix/Sprint8/6080/camdecmps/get-tee-feedback.sql (hunk 3 above)
Comment (djw4erg) [reply deployments/ecmps/release-v2.0-fix/Sprint8/6080/camdecmps/get-tee-feedback.sql:23]: Thanks for the clarification and I agree with your goal.  I am going to resolve this conversation.

**Review by djw4erg — APPROVED** [review]

**Pull request merged**